shared/js/sync/errors.js

```diff
@@ -0,0 +1,31 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+'use strict';
+
+(function(exports) {
+
+  var unrecoverableErrors = {
+    ERROR_REQUEST_SYNC_REGISTRATION: 'Cannot register sync request',
+    ERROR_GET_FXA_ASSERTION: 'Cannot get Firefox Accounts assertion',
+    ERROR_SYNC_REQUEST: 'Cannot perform sync request'
+  };
+
+  var recoverableErrors = {
+    ERROR_SYNC_APP_KILLED:
+      'The app was killed while performaning a sync operation',
+    ERROR_SYNC_APP_GENERIC: 'Error while trying to sync'
```

Comment:
We need to support L10N for these error messages as well, and errors.js only defines the L10N index/key.
If settings app will show these error messages, settings.en-US.properties would be like this with 'fxsync' prefix:
fxsync-error-request-sync-registration=Cannot register sync request
fxsync-error-get-fxa-assertion=Cannot get Firefox Accounts assertion
fxsync-error-sync-request=Cannot perform sync request
fxsync-error-sync-APP-killed=The app was killed while performaning a sync operation
fxsync-error-sync-APP-generic=Error while trying to sync